src/Storefront/Controller/AccountOrderController.php

```diff
@@ -110,16 +108,20 @@ public function orderSingleOverview(Request $request, SalesChannelContext $conte
             $page = $this->orderPageLoader->load($request, $context);
 
             $this->hook(new AccountOrderPageLoadedHook($page, $context));
-        } catch (GuestNotAuthenticatedException|WrongGuestCredentialsException|CustomerAuthThrottledException $exception) {
-            return $this->redirectToRoute(
-                'frontend.account.guest.login.page',
-                [
-                    'redirectTo' => 'frontend.account.order.single.page',
-                    'redirectParameters' => ['deepLinkCode' => $request->get('deepLinkCode')],
-                    'loginError' => ($exception instanceof WrongGuestCredentialsException),
-                    'waitTime' => ($exception instanceof CustomerAuthThrottledException) ? $exception->getWaitTime() : '',
-                ]
-            );
```

Comment:
Since I've never had to do something like this, not sure if this is the right approach. So open to commentary here.

Reply:
that is the reason why we keep specific exceptions if needed. imo the current change makes it more complicated to read. 

I would keep the specific exceptions. that is also what the ADR says: if we catch the exceptions in our code, a specific exception should stay

Reply:
But then that's even more of a reason to not use DomainExceptions. It gets messy. 

So to clean the PHPStan file, do you suggest I add an inline `// @phpstan` comment here?

Reply:
no, you have a method in the domain exception, that just returns the specific exception